From 8ca08c649436ed93ec24cb15f42523e1ffa45298 Mon Sep 17 00:00:00 2001
From: "Zhong,Fangjian" <fangjian.zhong@intel.com>
Date: Thu, 21 Jul 2022 17:34:05 +0800
Subject: [PATCH] [trusty][rpmb_dev] remove socket file if exist

When initializing rpmb_dev, a socket file would be created. However,
during CTS security test, socket file created might be not removed
successfully.

In order to enable socket file does not exist, this file should be
checked before created. If socket file exists, then remove it and
create a new one.
---
 trusty/utils/rpmb_dev/rpmb_dev.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/trusty/utils/rpmb_dev/rpmb_dev.c b/trusty/utils/rpmb_dev/rpmb_dev.c
index 202562114..512edfec1 100644
--- a/trusty/utils/rpmb_dev/rpmb_dev.c
+++ b/trusty/utils/rpmb_dev/rpmb_dev.c
@@ -585,6 +585,10 @@ int main(int argc, char** argv) {
         return EXIT_FAILURE;
     }
 
+    if (access(socket_path, F_OK)) {
+        remove(socket_path);
+    }
+
     /*
      * If the file is already initialized, exit early.
      */
-- 
2.25.1

